adam/language_specific/english/english_language_generator.py

```diff
@@ -606,8 +607,36 @@ def _preposition_for_region_as_goal(self, region: SituationRegion) -> str:
                 # TODO: put constraints on the axis
             ):
                 return "on"
+            elif region.distance == PROXIMAL and not region.direction:
+                return "to"
+            elif region.distance == INTERIOR:
+                return "in"
+            elif region.direction == GRAVITATIONAL_UP:
+                return "over"
             elif region.direction == GRAVITATIONAL_DOWN:
                 return "under"
+            elif region.direction and self.situation.axis_info:
```

Comment:
Does this `elif` actually get hit for cases w/ addressees? In the event that `region.distance` is specified (in all cases other than DISTAL), this statement won't be reached.

Reply:
Follow up question: is it necessarily always the case that things in-front-of or behind other things are necessarily PROXIMAL? If not - the additional check on line 610 isn't sufficient to make sure addressee situations are handled.

Reply:
1) Currently, this `elif` stack should be in proper order so that only the desired preposition is returned. Additionally, the addressee is only relevant information if we have failed to meet another condition for a preposition with the reference object. There is an addressee marked in all scenes regardless of if there is an object in the scene which is `FacingAddresseeAxis` 

2) The independent preposition curriculum defines "in front of" and "behind" as either `PROXIMAL` or `DISTAL` as I could say "the teacher is in front of my laptop" while I'm in the back of the classroom and the teacher is at the front and the preposition usage is still valid, even if not the most accurate to describe the location of the teacher. Likewise, I could say "my classmate is in front of my laptop" for a student directly in front of me one row up. (These could be defined as `PROXIMAL` and `DISTAL` in my understanding)

Reply:
Sounds good. I would recommend inserting a comment that notes why DISTAL doesn't need an entry in the elif stack for the benefit of future readers.

Reply:
I second @chris-jenkins-isi on the comment